File: yht0827/2월/2월 28일/프로세스.java
import java.util.Comparator;
import java.util.PriorityQueue;
import java.util.Queue;

class Solution {

	public int solution(int[] priorities, int location) {
		int answer = 0;

		Queue<Integer> pq = new PriorityQueue<>(Comparator.reverseOrder());

		for (int priority : priorities)
			pq.offer(priority);

		while (!pq.isEmpty()) {
			for (int i = 0; i < priorities.length; i++) {
				if (priorities[i] == pq.peek()) {
					if (i == location) {
						return ++answer;
					}
					pq.poll();
					answer++;
				}
			}
		}

		return answer;
	}
}

File: jeong-hyuk/2월28일/프로세스.py
def solution(priorities, location):
    answer = 0
    
    while True:
        front_doc = priorities.pop(0)
        location -= 1
        
        if any(front_doc < priority for priority in priorities):
            #우선순위가 더 높은 문서가 있다면, 맨 뒤로 이동
            priorities.append(front_doc)
            if location < 0:
                location = len(priorities) - 1
        else:
            #우선순위가 가장 높은 문서라면 출력
            answer += 1
            if location == -1:
                break
    
    return answer